simulation/groups_interface.py

```diff
@@ -16,9 +16,16 @@ class PopulationGroup(metaclass=ABCMeta):
     id = count(0)
```

Comment:
Do we still need this `id`? If not, maybe it is better to clean up these things.